### PR TITLE
Removing CONFIG_PINCTRL=y from the nxp boards

### PR DESCRIPTION
**Pull request by EmilioCBen**
This PR addresses the NXP portion of [78619](https://github.com/zephyrproject-rtos/zephyr/issues/78619).

### CODE REVIEW

**Review by nordicjm — DISMISSED** [review]

**Review by decsny — DISMISSED** [review]
is this meant to be a draft? where are the driver kconfig changes?

**Comment by EmilioCBen** [comment]
@nordicjm 
Hey, it seems that this PR may be ready but as you point out [here](https://github.com/zephyrproject-rtos/zephyr/issues/78353#issuecomment-2348072739) I should move the Board's Enablement of PINCTRL from 
*_defconfig to the Boards Kconfig, this seems to break Check Compliance. Otherwise Check Compliance will pass.
What do think on the matter?

Also please let me know if I should make any additional changes.

File: boards/nxp/frdm_kl25z/Kconfig.frdm_kl25z [hunk 1]
```diff
@@ -4,3 +4,4 @@
 config BOARD_FRDM_KL25Z
 	select SOC_MKL25Z4
 	select SOC_PART_NUMBER_MKL25Z128VLK4
+	select PINCTRL
```
Comment (nordicjm) [inline boards/nxp/frdm_kl25z/Kconfig.frdm_kl25z:7]: this does not go here

File: boards/nxp/twr_ke18f/Kconfig.twr_ke18f [hunk 2]
```diff
@@ -4,3 +4,4 @@
 config BOARD_TWR_KE18F
 	select SOC_MKE18F16
 	select SOC_PART_NUMBER_MKE18F512VLL16
+	select PINCTRL
```
Comment (nordicjm) [inline boards/nxp/twr_ke18f/Kconfig.twr_ke18f:7]: nor here

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: boards/nxp/frdm_kl25z/Kconfig.frdm_kl25z (hunk 1 above)
Comment (EmilioCBen) [reply boards/nxp/frdm_kl25z/Kconfig.frdm_kl25z:7]: Where does this belong if the board is using the pinctrl api?

Comment (nordicjm) [reply boards/nxp/frdm_kl25z/Kconfig.frdm_kl25z:7]: Where is the .c file in the board folder where pinctrl is used?

Comment (EmilioCBen) [reply boards/nxp/frdm_kl25z/Kconfig.frdm_kl25z:7]: Ah, I apologize for the confusion, somehow I thought I saw a pinctrl API in this directory, but there is one for the twr board so the question still applies, the select for this board though I can remove on the next update.

File: boards/nxp/twr_ke18f/Kconfig.twr_ke18f (hunk 2 above)
Comment (EmilioCBen) [reply boards/nxp/twr_ke18f/Kconfig.twr_ke18f:7]: https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/nxp/twr_ke18f/pinmux.c

Comment (nordicjm) [reply boards/nxp/twr_ke18f/Kconfig.twr_ke18f:7]: this one is acceptable

**Comment by dleach02** [comment]
@EmilioCBen please address CI failures. 

Also "meant to fix" should be "fixes #<number>"

File: boards/nxp/twr_ke18f/Kconfig.twr_ke18f (hunk 2 above)
Comment (nordicjm) [reply boards/nxp/twr_ke18f/Kconfig.twr_ke18f:7]: It needs to go into `Kconfig`, not `Kconfig.<board_name>`

**Comment by EmilioCBen** [comment]
> @EmilioCBen please address CI failures.
> 
> Also "meant to fix" should be "fixes #"

Updated

File: boards/nxp/twr_ke18f/Kconfig.twr_ke18f (hunk 2 above)
Comment (EmilioCBen) [reply boards/nxp/twr_ke18f/Kconfig.twr_ke18f]: Thank you, appreciate the help!

**Comment by EmilioCBen** [comment]
@nordicjm 
Thank you for the help on this,
I believed I addressed your comments would you mind taking another look

**Review by nordicjm — APPROVED** [review]
Thanks!

**Review by dleach02 — APPROVED** [review]

**Review by decsny — APPROVED** [review]

**Pull request merged**